### PR TITLE
Add first tests for tonation and sound generation

### PR DESCRIPTION
**Pull request by JakubBilski**
Tests are now in main catalogue, bc I was unable to import music from src/tests/test_data.py (Kuba pls help, I remember we did it like a year ago in some project). If imports can be fixed, we would like all test_data, main_tonation_from_sounds_test and main_sound_from_file_test in src/test catalogue

Both tonations and sounds tests are based on a idea of match factor: we compute how much of a returned timeline matches the model timeline. For example, if we have

image

then, the match factor would be 1-((26-24)+(40-32))/40=0.75,
and Mismatches (printed to the user in verbose modes) would be
24-26, expected blue, got green
32-40, expected blue, got red

Now, we treat None at the same terms as any other note. We can discuss it.
This approach works for one-note-at-a-time problems. For many notes at the same time a different (but maybe similar) approach must be implemented.

Test cases for sounds are basically copied results from the current algorithm (rounded to the third decimal place, that causes a small error). They are not ideal, I will delete some of the errors from there later. As well, much more test cases need to be described in code, since I only did these from data\other_rec

image

Tonations, on the other hand, are real, and an error can be seen

image

The same with verbose settings

image

### CODE REVIEW

**Comment by AnnaMBuchman** [comment]
ach śpij C-dur nie c-moll -> f-moll ma 4 bemole które w ogóle nie występują
jedzie pociąg z daleka-> to F-dur ponieważ 1. kończy się na F i ma bemol, nie wiem czy w algorytmie mamy utory nie zaczynające się od toniki, ale są takie ale ten się zaczyna od toniki tylko w 2 przewrocie tz. C F A (bazowy to F A C ;) )
o mój rozmarynie głownie opiera się na gamie
Śmiesznie że przybieżeli według naszego algorytmu jest molowe :P

**Comment by JakubBrojacz** [comment]
Przetwory w tej chwili nie mają znaczenia dla algorytmu. 
Mógłbyś wygenerować lepsze nuty korzystając z algorytmu z mojego brancha (notes duration), możemy pogadać o tym jutro.

File: src/main_sound_from_file_test.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        timestamp = next_timestamp
+
+    mismatches_duration = sum([m.duration for m in mismatches])
+    if duration != duration_model:
+        mismatches_duration += abs(duration - duration_model)
+    return 1-mismatches_duration/duration, mismatches
+
+if __name__ == "__main__":
+    verbose_factor_threshold = parse_args().verbose_factor_threshold
+    verbose_duration_threshold = parse_args().verbose_duration_threshold
```
Comment (JakubBrojacz) [inline src/main_sound_from_file_test.py:66]: It is better to invoke parse_args() just once, eg
```python
args = parse_args()
verbose_factor_threshold = args.verbose_factor_threshold
verbose_duration_threshold = args.verbose_duration_threshold
```

**Review by JakubBrojacz — APPROVED** [review]

File: src/main_sound_from_file_test.py (hunk 1 above)
Comment (JakubBilski) [reply src/main_sound_from_file_test.py]: Done

**Pull request merged**